### ISSUE TITLE
Manage many bots with just one gtp2ogs command !!

### ISSUE THREAD

**Issue by wonderingabout** [online-go/gtp2ogs#102]
so

while i was answering here https://github.com/online-go/gtp2ogs/pulls/101
i just got a very important idea of adding support to run several bots with just on gtp2ogs command

my purpose for doing this is for example if we replay an AI tournament, or even just for live games, i want my GPU to play 1 move at a time, yet spread the moves evely between bots

for example

bot 1

game a
game b
game c
bot 2
game z
game y
i want gtp2ogs to handle my gpu power if "one game at a time is specified"

for example

play first for bot 1 game a
then play for bot 2, game z
then play for bot 1, game b
etc...

@roy7 @Dorus @windo @lemonsqueeze

**Comment by roy7** [comment]
I feel like this would be a huge undertaking. :) Perhaps something like https://github.com/online-go/gtp2ogs/pull/64 could be done to process games one at a time across processes but this isn't something I'd try to tackle in gtp2ogs myself.

**Comment by roy7** [comment]
Anything is possible, it's just a question of the manhours to implement. Added complexity also can add bugs and unintended consequences. You'd also need to redo the way the command lines work or add support for an extra config file since you'd have multiple usernames, API keys, bot commands, etc.

**Comment by wonderingabout** [comment]
oh i see @roy7 

i thought more on the angle of cpu thread management

but actually what seems the longest is software javascript connection manage

but still, it doesnt seem that hard
i hope i can get another look at it when i'm mroe skilled :)

**Comment by Dorus** [comment]
Tbh if you want to jump into something like this we really need to do some ground work first, like splitting the code over multiple files, identify a few classes and preferably also some way to control the configuration while the program is running. The current program is already growing past what you can realistically put in one file.

**Comment by windo** [comment]
If the main motivation is to limit the number of total games played, then we could also just implement that specific feature by having multiple `gtp2ogs` synchronize with each other. That reduced scope would make the feature much more tractable.

**Comment by TokumotoK** [comment]
Either way is a big benefit for me.  I have been running 3 copies of gtp2ogs on a single Windows box.

**Comment by wonderingabout** [comment]
yes i feel same as @TokumotoK , either would be great

but i'm actually very curious and excited about @Dorus 's idea

**Comment by wonderingabout** [comment]
as it looks quite fantasy for now, closing it for now, and me or someone else may reopen it in some future time if we get more concrete ideas how to do this if we still really want :+1:

**Comment by windo** [comment]
I also run multiple gtp2ogs instances in parallel and it isn't quite annoying enough to start working on improving the situation.

I think in general you could already almost instantiate multiple `Connection` objects - the trouble is that there is a single config. I think you would effectively want to be able to configure the different bots separately - starting from command line but also possibly different time options or rank restrictions, etc.

For that, you almost certainly need a configuration file where to define per-bot options in per-bot sections. If there was a config file, then I think it would be just a matter of gluing all the existing things together the right way.

**Comment by wonderingabout** [comment]
> For that, you almost certainly need a configuration file where to define per-bot options in per-bot sections. If there was a config file, then I think it would be just a matter of gluing all the existing things together the right way.

i put emphasis on that last line, we really would want to have a config file regardless of this issue, so i'll reopen an issue to remember it if not already done

as for this issue, the matter is not urgent i think, so i'll reopen an issue for config file :+1:

**Comment by wonderingabout** [comment]
the config file issue was actually here #125 

so config file discussion and advancements can be continued there
